### ISSUE TITLE
How to understand if I have soldered the crystal properly?

### ISSUE THREAD

**Issue by ybelakov** [Nickduino/Pi-Somfy#106]
After soldering I took my multimeter and tested every path on the emitter (continuity mode) and it seems that the paths are the same as in the original emitter (before desoldering). However it is not working. I checked my wiring 100 times and I am sure it is correct. I checked the jumper cables with the multimeter, too (they are OK). Any suggestions?

May be I am wrong in the setup:
1) I start the web server
2) Add new shutter
3) Click the program button on the back of the Somfy remote and click "Try programming again". The shutter never moves except when using the remote.

**Comment by yavinmaster** [comment]
Do you have any normal RF switches you could test?  I tested my transmitter with https://github.com/milaq/rpi-rf and a RF switch before soldering on the new crystal.  I have just gone through the full setup and it all worked for me.

**Comment by Nickduino** [comment]
You could probably do the test with the new oscillator, it's unlikely your receive is as picky as Somfy's frequency-wise

**Comment by ybelakov** [comment]
So you are basically suggesting something like this:
1) Connect both the RF receiver and RF emitter to the RPI 
2) Try sending signals from the emitter to the receiver and see if it works.

I still have one untouched emitter (433.92) which I can test with.
Did I understand you, guys, correctly?

**Comment by Nickduino** [comment]
You would need another RECEIVER, not emitter. I don't know if you can run them both on the Pi at the same time but you could try. On the top of my head, I think there is a Python port of VirtualWire.

The other solution is to use two Pis or a Pi and an arduino (which I did when first learning the radio stuff)

**Comment by ybelakov** [comment]
Hi, 
Thanks for the respond!

As far as I have understood this on the left is the emitter and this on the right is the receiver:
![image](https://user-images.githubusercontent.com/60155068/107886740-a0057380-6f0a-11eb-8c6e-2a9a718091d8.png)

I will try connecting them both to the same RPI (I have only one) and see if it works. Here's a wiring diagram I will be using if somebody's interested in testing, too.
![image](https://user-images.githubusercontent.com/60155068/107886787-dba03d80-6f0a-11eb-9d5b-8b6982aba00d.png)

**Comment by Nickduino** [comment]
The long one is your receiver, correct

**Comment by alonjr** [comment]
@ybelakov I assume that you found the problem. Can you share with us what it was and how you resolved it. I am also struggling to get the solution work with my somfy powered awning.
Tnx

**Comment by Nickduino** [comment]
>  my somfy powered awning

Are you sure they're using the RTS protocol?

**Comment by Nickduino** [comment]
No, it can be io on the more recent hardware

**Comment by alonjr** [comment]
@Nickduino I believe that you are correct!
The user manual of the remote control shows  the model name as _**Situo 1 io**_. 
In that case, perhaps this is the reason I am not able to get Pi-Somfy to work.
Is there a solution that you are aware of that might work for me?
tnx

**Comment by Nickduino** [comment]
io is a more modern and safer two-way protocol. You'll need one of the two Somfy boxes (or possibly a third party that does io?) to control your stuff

**Comment by alonjr** [comment]
Thanks. While I will not have use of your project, at least I learned something new and had fun doing so.



From: Nickduino ***@***.***>
Reply-To: Nickduino/Pi-Somfy ***@***.***>
Date: Monday, August 2, 2021 at 18:42
To: Nickduino/Pi-Somfy ***@***.***>
Cc: alon rosenfeld ***@***.***>, Comment ***@***.***>
Subject: Re: [Nickduino/Pi-Somfy] How to understand if I have soldered the crystal properly? (#106)


io is a more modern and safer two-way protocol. You'll need one of the two Somfy boxes (or possibly a third party that does io?) to control your stuff

**Comment by Nickduino** [comment]
Great! Good luck with the rest

**Comment by sfla** [comment]
I too am struggling with this (and I'm sure they use RTS). I don't have any fancy meters to check the power, and I'm completely new to both raspberry pi and soldering. Is there any form of software I can use to monitor or query the connection to the emitter to see if it's at least recognized as a connection?

**Comment by Nickduino** [comment]
> Is there any form of software I can use to monitor or query the connection to the emitter to see if it's at least recognized as a connection?

It's not a peripheral that needs a driver or something, it doesn't need to be "recognized": the RPi doesn't know there an emitter (or nothing) connected there.

Now if you want to make sure your soldering is good and your wiring correct, your best chance is to make the system work with your blinds.

**Comment by hourliert** [comment]
Hi,
I just went through the setup this afternoon. My first transmitter was screwed and I only realized it after desoldering + soldering everything. Had to desolder + resolder another one, glad I bought 5.
I have used a multimeter to debug it and essentially making sure that: the ground pin connect with the ground of the crystal, the 5V pin connects with the top pin of the crystal, and any other links are still working fine.

It's now working like a charm, super happy! Thanks again!
